### PR TITLE
Adapters

### PR DESCRIPTION
**Pull request by vinbo8**
Add support for adapters, engage using `--add-adapters`.

### CODE REVIEW

**Review by gmastrapas — COMMENTED** [review]
`adapters` needs to be in the requirements file

**Review by gmastrapas — COMMENTED** [review]
We also need a run script to test the adapters at `scripts/`

File: src/open_clip/factory.py [hunk 1]
```diff
@@ -317,6 +320,12 @@ def _convert_ln(m):
         force_preprocess_cfg['size'] = model.visual.image_size
     set_model_preprocess_cfg(model, merge_preprocess_dict(preprocess_cfg, force_preprocess_cfg))
 
+    if add_adapters:
+        adapters.init(model.text.transformer)
+        adapter_config = AdapterConfig.load("seq_bn")
+        model.text.transformer.add_adapter("test_adapter", adapter_config)
```
Comment (gmastrapas) [inline src/open_clip/factory.py]: So the model from OpenCLIP already has a builtin method called `add_adapter`?

File: src/open_clip/factory.py [hunk 2]
```diff
@@ -317,6 +320,12 @@ def _convert_ln(m):
         force_preprocess_cfg['size'] = model.visual.image_size
     set_model_preprocess_cfg(model, merge_preprocess_dict(preprocess_cfg, force_preprocess_cfg))
 
+    if add_adapters:
+        adapters.init(model.text.transformer)
+        adapter_config = AdapterConfig.load("seq_bn")
```
Comment (gmastrapas) [inline src/open_clip/factory.py]: I would add the adapter name to the command line arguments

**Review by gmastrapas — CHANGES_REQUESTED** [review]

File: src/open_clip/factory.py (hunk 1 above)
Comment (vinbo8) [reply src/open_clip/factory.py]: No, it's from the [PEFT](https://github.com/huggingface/peft) library, that adds support for this stuff to `transformers`.

**Pull request closed without merging**